src/swarm/Swarm.ts

```diff
@@ -144,14 +144,61 @@ const uploadData = async (data: string, swarmGateway: string): Promise<string> =
     return text;
 };
 
-const downloadData = async (hash: string, timeout: number, swarmGateway: string): Promise<string> => {
+const uploadUint8Array = async (data: Uint8Array, swarmGateway: string): Promise<string> => {
+    Debug.log('uploadUint8Array', 'data.length', data.length);
+    const url = swarmGateway + defaultUrlScheme;
+    const options: RequestInit = {
+        headers: {
+            'Content-Type': 'text/plain',
+        },
+        method: 'POST',
+    };
+    options.body = data;
+    const response = await safeFetch(url, options);
+    const text = await response.text();
+    return text;
+};
+
+const downloadString = async (hash: string, timeout: number, swarmGateway: string): Promise<string> => {
     const url = swarmGateway + '/bzz:/' + hash + '/';
     Debug.log('downloadData:', url);
     const response = await safeFetchWithTimeout(url, undefined, timeout);
     const text = await response.text();
     return text;
 };
 
+const fetchArrayBuffer = (url: string, timeout: number): Promise<Uint8Array> => {
+    return new Promise((resolve, reject) => {
+        const request = new XMLHttpRequest();
```

Comment:
why are we using here XMLHttpRequest and not fetch?

Reply:
React Native did not implement fetch with ArrayBuffer type and they consider this a feature request (which they don't want to implement) instead of a bug:

https://github.com/facebook/react-native/issues/9947

I found the workaround here:
https://stackoverflow.com/questions/43711977/how-to-use-fetch-api-with-arraybuffer-response-type-and-react-native